teps/0066-dogfooding-tekton.md

```diff
@@ -0,0 +1,320 @@
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+The Tekton community has been "dogfooding" Tekton since the early stages of the
+project.
+
+> The term "dogfooding" is an IT slang for the use of one's own products.
+> In some uses, it implies that developers or companies are using their own products
+> to work out bugs, as in beta testing. One benefit of dogfooding is that it shows
+> that a company is confident about its products.
+
+We would like to capture the goals of the dogfooding effort in the Tekton community,
+as document guiding principles, existing achievements as well as future roadmap.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals and non-goals of
+this TEP.  Describe why the change is important and the benefits to users.  The
+motivation section can optionally provide links to [experience reports][] to
+demonstrate the interest in a TEP within the wider Tekton community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+The dogfooding effort in Tekton has been going on since the early stages of the project,
+so why writing a TEP now?
+
+When the dogfooding work was started we did not have a TEP process in place, we relied on
+[shared Google documents](https://docs.google.com/document/d/1Tf3tdIBwkN5kxMuyYMo802F-DyOdsl_bgh-78a9prWQ/edit#)
+instead. This TEP captures the work done this far, its guiding principles, the future
+roadmap; it aims to raise awareness about this work in the community, and to make it
+easier for new developers to get started and contribute to it.
+
+### Goals
+
+Tekton is non-opinionated by design. It allows designing CI/CD pipelines and systems,
+but it's not prescriptive about how to do that. The dogfooding effort offers CI/CD
+services to the Tekton community through Tekton, and in doing so it must take design
+decisions on how to build such services through Tekton. Goals of this TEP are to:
+
+- Formalize the motivation for the dogfooding effort
+- Describe the design principles for CI/CD services based on Tekton
+- Identify the services implemented through Tekton
+- Identify areas were more work is needed and define a future roadmap
+
+### Non-Goals
+
+This TEP does not aim to design specific services based on Tekton, that work will be tracked
+in dedicated TEPs where needed.
+
+### Use Cases (optional)
+
+Use cases that the dogfooding effort implements are:
+
+- Experience Tekton from the operator, author and user point of view, which allows the
+  Tekton community to:
+  - Identify missing functional features and usability issues
+  - Identify operational problems and security concerns
+  - Discover bugs that can be better identified in a running system, such as upgrade issues,
+    a problems related to maintainability and scale
```

Comment:
Does something like this fit in this list.

`- Define/Document what a stable state of Tekton Pipelines/Triggers... installation on a cluster is.`

In other words, How can a user/operator ensure that an installation/upgrade is successful especially when `kubectl apply` does not report an error. (diagnostics, how to check state of cluster)

Reply:
Do you mean during further validation *before* a release? 
We run E2E tests on every commit of pipeline and triggers but what that does not verify is:

- upgrade from a running system, upgrade of tekton resources
- running at scale / under load
- integration between pipeline / triggers

That's what my bullet point `Discover bugs that can be better identified in a running system, such as upgrade issues,  and problems related to maintainability and scale` is about. 

What is missing is having enough resources and load on the `robocat` cluster. Today we only install the nightlies and that's it, the only service we run from robocat is deployment of full releases to `dogfooding` but that's triggered manually when new releases are available.

Reply:
I added an extra use case to the least, I hope it addresses your point.